src/core/i18n/useMessages.ts

```diff
@@ -1,6 +1,6 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { useIntl } from 'react-intl';
+import { IntlShape, useIntl } from 'react-intl';
```

Comment:
I'm just curious about the introduction of `IntlShape ` in this refactor. Could you share a bit about the reasoning behind it and what advantages it brings? Always great to learn from these kinds of improvements!

Reply:
`IntlShape` is a type. It is the return type of `useIntl()`. Previously we've been able to rely on type inference, but because I moved `injectIntl()` to a separate function and needed to accept `intl` as a parameter, I needed to specify it's type.